src/java/org/apache/cassandra/config/DatabaseDescriptor.java

```diff
@@ -1459,20 +1461,23 @@ public static void applyPartitioner(Config conf)
         partitionerName = partitioner.getClass().getCanonicalName();
     }
 
-    private static DiskAccessMode resolveCommitLogWriteDiskAccessMode(DiskAccessMode providedDiskAccessMode)
+    private static Pair<DiskAccessMode, Boolean> resolveCommitLogWriteDiskAccessMode(DiskAccessMode providedDiskAccessMode)
     {
         boolean compressOrEncrypt = getCommitLogCompression() != null || (getEncryptionContext() != null && getEncryptionContext().isEnabled());
         boolean directIOSupported = false;
         try
         {
-            String commitLogLocation = getCommitLogLocation();
+            if (COMMITLOG_DIRECTIO_SUPPORT_FOR_TEST.getBoolean())
```

Comment:
We shouldn't introduce changes in the logic for tests. We can achieve this with Byteman, by adding a rule to return 0 for `FileUtils.getBlockSize`. See for example [CompactionsBytemanTest](https://github.com/apache/cassandra/blob/trunk/test/unit/org/apache/cassandra/db/compaction/CompactionsBytemanTest.java).